provider/file/file_test.go

```diff
@@ -0,0 +1,193 @@
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/bank-vaults/secret-init/provider"
+)
+
+func TestMain(m *testing.M) {
+	exitCode := m.Run()
+
+	// teardown environment variables after tests are done
+	teardownEnvs()
+
+	os.Exit(exitCode)
+}
+
+func TestNewFileProvider(t *testing.T) {
+	// create a new secret file and write secrets into it
+	tmpfile := createTempFileWithContent(t)
+	defer os.Remove(tmpfile.Name())
+
+	// create new environment variables
+	// for file-path and secrets to get
+	setupEnvs(t, tmpfile)
+
+	var fileProvider provider.Provider
+	providerName := os.Getenv("PROVIDER")
+	switch providerName {
+	case "file":
+		newFileProvider, err := NewFileProvider(os.Getenv("SECRETS_FILE_PATH"))
+		fileProvider = newFileProvider
+		if err != nil {
+			t.Fatal(err)
+		}
+	default:
+		t.Fatalf("invalid provider specified: %s", providerName)
+	}
```

Comment:
When testing, do not copy-paste the code, but call the functions/methods you are testing, look around in other BV repos for inspiration! 🙂